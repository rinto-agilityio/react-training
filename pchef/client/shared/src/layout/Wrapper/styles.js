// Lib
import { StyleSheet } from 'react-native'

// Themes
import { METRICS } from '../../themes'

// Create style for layout
const styles = StyleSheet.create({
  container: {
    ...METRICS.flexCenter,
    display: 'flex',
    flexWrap: 'wrap',
  },
  column: {
    flexDirection: 'column',
  },
  row: {
    flexDirection: 'row',
  },
  left: {
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  },
  right: {
    alignItems: 'flex-end',
    justifyContent: 'flex-start',
  },
  middle: {
    alignItems: 'flex-start',
    justifyContent: 'space-between',
  },
  spaceAround: {
    justifyContent: 'space-around',
  },
  spaceBetween: {
    justifyContent: 'space-between',
  },
})

export default styles
