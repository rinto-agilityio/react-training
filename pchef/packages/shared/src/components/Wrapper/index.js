// libs
import React from 'react';
import { View, Text } from 'react-native';

type Props = {
  width: number,
  height: number,
  children: React$Element<any>,
  customStyle: Object
};

const Wrapper = ({
  width,
  height,
  children,
  customStyle
}: Props) => (
  <View style={[
    {
      width: width,
      height: height,
    },
    customStyle
  ]}>
    {children}
  </View>
);

export default Wrapper;

Wrapper.defaultProps = {
  width: 200,
  height: 200,
  customStyle: {}
};
