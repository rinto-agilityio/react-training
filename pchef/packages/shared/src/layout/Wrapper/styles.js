// Lib
import { StyleSheet } from 'react-native'

// Create style for layout
const styles = StyleSheet.create({
  container: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexWrap: 'wrap',
  },
  column: {
    flexDirection: 'column',
  },
  row: {
    flexDirection: 'row',
  },
  left: {
    alignItems: 'flex-start',
    justifyContent: 'flex-end',
  },
  right: {
    alignItems: 'flex-end',
    justifyContent: 'flex-start',
  },
  spaceAround: {
    justifyContent: 'space-around',
  },
  spaceBetween: {
    justifyContent: 'space-between',
  },
})

export default styles
