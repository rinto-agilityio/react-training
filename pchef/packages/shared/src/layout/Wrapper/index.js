// Libs
import * as React from 'react'
import { View } from 'react-native'

// Styles
import styles from './styles'

type Props = {
  children: React.Node,
  direction?: string,
  childPosition?: string,
  flexGrow?: number,
  customStyles?: {} | Array<{}>,
  onLayout?: () => void,
}

const Wrapper = ({
  children,
  direction,
  childPosition,
  flexGrow,
  customStyles,
  onLayout,
}: Props) => (
  <View
    onLayout={onLayout}
    style={[
      styles.container,
      styles[direction],
      styles[childPosition],
      customStyles,
      { flexGrow },
    ]}
  >
    {children}
  </View>
)

Wrapper.defaultProps = {
  direction: 'row',
  childPosition: 'spaceAround',
  flexGrow: 0,
  customStyles: {},
  onLayout: () => {},
}

export default Wrapper
