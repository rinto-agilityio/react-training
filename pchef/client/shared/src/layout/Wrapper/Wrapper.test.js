// Libs
import React from 'react'
import renderer from 'react-test-renderer'

// Components
import Wrapper from '.'

const props = {
  onLayout: jest.fn(),
}

const renderComponent = renderer.create(
  <Wrapper {...props}>
    <p>test</p>
  </Wrapper>,
)

describe('Components', () => {
  describe('Wrapper', () => {
    let component

    beforeEach(() => {
      component = shallow(
        <Wrapper {...props}>
          <p>test</p>
        </Wrapper>,
      )
    })
    it('should render Button component', () => {
      expect(renderComponent).toMatchSnapshot()
    })

    it('should have call onLayout', () => {
      component.props().onLayout()
      expect(props.onLayout).toHaveBeenCalled()
    })
  })
})
