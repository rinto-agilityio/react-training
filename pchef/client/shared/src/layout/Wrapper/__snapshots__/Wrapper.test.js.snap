// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Wrapper should render Button component 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flexDirection-18u37iz r-flexWrap-1w6e6rj r-justifyContent-a2tzq0"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "WebkitFlexGrow": 0,
      "flexGrow": 0,
      "msFlexPositive": 0,
    }
  }
>
  <p>
    test
  </p>
</div>
`;
